code/game/objects/items/devices/lightreplacer.dm

```diff
@@ -98,6 +98,33 @@
 			user << "<span class='warning'>You need a working light!</span>"
 			return
 
+	if(istype(W, /obj/item/weapon/storage/box/lights))

```

Comment:
you should make this ```/obj/item/weapon/storage``` or atleast ```/box```, so you can fill it from any container/any box.